src/hooks/hipchat.js

```diff
@@ -0,0 +1,52 @@
+const Hook = require("../hook");
+
+const MXControlEvent = require("../events/mxcontrol");
+const MXControlHandler = require("../handlers/mxcontrol");
+const minimist = require("minimist");
+
+//a primer to this type of function
+//(cause this shit is difficult the first time)
+//I'm definining a function here that takes in a trigger,
+//and returns the result of another function which does some destructuring
+//when called on the given trigger, which is passed into said function
+//w/ destructuring on the last line of this function, after the return of the
+//inner function.
+
+const parseHipchatTrigger = trigger =>
+    (({
+        event,
+        item: {
+            message: {
+                message,
+                from: {
+                    id,
+                    mentionName: mention_name,
+                    name
+                }
+            }
+        }
+    }) => {
+        const events = [];
+        let command, args;
+
+        if (event == "room_message") {
```

Comment:
you can avoid having to forward declare the variables by converting this fence to a hurdle